synopsys-task/src/synopsys-task/tools-parameter.ts

```diff
@@ -259,6 +275,44 @@ export class SynopsysToolsParameter {
       },
     };
 
+    const azureRepositoryName =
+      taskLib.getVariable(AZURE_ENVIRONMENT_VARIABLES.AZURE_REPOSITORY) || "";
+
+    if (inputs.COVERITY_PROJECT_NAME) {
+      covData.data.coverity.connect.project.name = inputs.COVERITY_PROJECT_NAME;
+    } else {
+      covData.data.coverity.connect.project.name = azureRepositoryName;
+    }
+
+    if (inputs.COVERITY_STREAM_NAME) {
+      covData.data.coverity.connect.stream.name = inputs.COVERITY_STREAM_NAME;
+    } else {
+      const buildReason =
+        taskLib.getVariable(AZURE_ENVIRONMENT_VARIABLES.AZURE_BUILD_REASON) ||
+        "";
+      if (buildReason == "PullRequest") {
```

Comment:
Instead of String "PullRequest" string can we make enum like

enum AzureBuildReason {
    Manual = 'Manual',
    IndividualCI = 'IndividualCI',
    BatchedCI = 'BatchedCI',
    Schedule = 'Schedule',
    ValidateShelveset = 'ValidateShelveset',
    CheckInShelveset = 'CheckInShelveset',
    PullRequest = 'PullRequest',
    BuildCompletion = 'BuildCompletion',
    ResourceTrigger = 'ResourceTrigger'
}

Reply:
Added the enum but used only PULL_REQUEST as for now others are not required